lib/Controller/ApiController.php

```diff
@@ -310,15 +310,15 @@ public function addParticipantToRoom($roomId, $newParticipant) {
 		}
 
 		if ($room->getType() === Room::ONE_TO_ONE_CALL) {
-			// In case a user is added to a one2one call, we create a new group call and add the participants manually
-			$room = $this->manager->createRoom(Room::GROUP_CALL, $this->secureRandom->generate(12));
-			foreach ($participants as $participant => $lastPing) {
-				$user = $this->userManager->get($participant);
-				if ($user instanceof IUser) {
-					$room->addUser($user);
-				}
+			// In case a user is added to a one2one call, we upgrade the call to a group call
+			try {
+				$room = $this->manager->upgradeRoom($room);
+			} catch (RoomNotFoundException $e) {
+				return new JSONResponse([], Http::STATUS_NOT_FOUND);
```

Comment:
How can this happen? The room has already been checked for existence in https://github.com/nextcloud/spreed/blob/99d9951440e311b4cec3b67be673e3af12a697d5/lib/Controller/ApiController.php#L297